From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Fri, 8 May 2020 09:55:15 +0200
Subject: ANDROID: fix merge issue in fsnotify.h

Youling reports that the fsnotify.h fixup done in 58dc0ef35a07
("ANDROID: Remove out-of-tree difference in fsnotify.h") might not have
been completly correct.

So try to fix this up by passing down lower_path in the call to
fsnotify_parent() in fsnotify_open().

Fixes: 58dc0ef35a07 ("ANDROID: Remove out-of-tree difference in fsnotify.h")
Cc: Matthias Maennich <maennich@google.com>
Cc: Daniel Rosenberg <drosen@google.com>
Reported-by: youling 257 <youling257@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I23b5cb2d41b82e386a4f0620efec6b75e5b2d138
---
 include/linux/fsnotify.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/fsnotify.h b/include/linux/fsnotify.h
--- a/include/linux/fsnotify.h
+++ b/include/linux/fsnotify.h
@@ -271,7 +271,8 @@ static inline void fsnotify_open(struct file *file)
 
 	if (path->dentry->d_op && path->dentry->d_op->d_canonical_path) {
 		path->dentry->d_op->d_canonical_path(path, &lower_path);
-		fsnotify_parent(lower_path.dentry, mask, NULL, FSNOTIFY_EVENT_PATH);
+		fsnotify_parent(lower_path.dentry, mask, &lower_path,
+				FSNOTIFY_EVENT_PATH);
 		fsnotify(lower_path.dentry->d_inode, mask, &lower_path, FSNOTIFY_EVENT_PATH, NULL, 0);
 		path_put(&lower_path);
 	}
